plotly_resampler/figure_resampler/figure_resampler.py

```diff
@@ -160,6 +162,15 @@ def show_dash(
         self._host = kwargs.get("host", "127.0.0.1")
         self._port = kwargs.get("port", "8050")
 
+        # check if self._port is already in use
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            if s.connect_ex(('localhost', int(self._port))) == 0:
+                error_message = "Address already in use\n" \
+                    f"Port {self._port} is in use by another program. " \
+                    "Either identify and stop that program, or start the server with a different port."
+                print(error_message)
```

Comment:
Why is print sys.exit used? Isn't it clearer to raise an error with that error message?

Reply:
I think the other real reason I dont want to throw an error here is that this is just calling the upstream libraries and honestly shouldnt be this library's problem to catch that imo as all it does is resample really well.  I will do some digging at some point today to confirm any assumptions in the above comment

Reply:
I'll have a look myself later this week (I guess we should look at dash & jupyter dash)  :)

Would be amazing if this PR avoids the cases where it takes dash way too long to notify the users when the port is already used!

Reply:
I raised the error and for reference here is where it is implemented in werkzeug, which is the error we were/are seeing with this change